sdk/python/feast/diff/infra_diff.py

```diff
@@ -36,15 +38,29 @@ def __init__(self):
         self.infra_object_diffs = []
 
     def update(self):
-        pass
+        """Apply the infrastructure changes specified in this object."""
+        for infra_object_diff in self.infra_object_diffs:
+            if infra_object_diff.transition_type in [
+                TransitionType.DELETE,
+                TransitionType.UPDATE,
+            ]:
+                infra_object = InfraObject.from_proto(
+                    infra_object_diff.current_infra_object
+                )
+                infra_object.teardown()
```

Comment:
why does an "UPDATE" cause a teardown?

Reply:
ah because an update transition is destructive: it needs to teardown the old infra and create the new one

as an example, consider changing the namespace of a datastore table: the table in the previous namespace should be deleted, and a new table in the new namespace should be created

an action item is to more clearly distinguish between destructive and non-destructive field changes - right now if any field is changed, it will be a destructive update

Reply:
got it that makes sense.